Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -91,24 +99,37 @@ void GridMasonryLayout::addItemsToFirstTrack(const HashMap<RenderBox*, GridArea>
         insertIntoGridAndLayoutItem(item, gridArea);
 }
 
+void GridMasonryLayout::placeItemsUsingOrderModifiedDocumentOrder()
+{
+    for (auto* child = m_renderGrid.currentGrid().orderIterator().first(); child; child = m_renderGrid.currentGrid().orderIterator().next()) {
+        if (m_renderGrid.currentGrid().orderIterator().shouldSkipChild(*child) || m_firstTrackItems.contains(child))
+            continue;
+        if (hasDefiniteGridAxisPosition(child, gridAxisDirection()))
+            insertIntoGridAndLayoutItem(child, masonryGridAreaFromGridAxisSpan(gridAxisSpanForItemWithDefiniteGridAxisPosition(*child)));
```

Comment:
It's a bit strange that in this case you have to call `masonryGridAreaFromGridAxisSpan`, while in the other case below it's already called by `positionForIndefiniteGridAxisItem`.

Why not rename `gridAxisSpanForItemWithDefiniteGridAxisPosition` to `positionForDefiniteGridAxisItem` and let it call `masonryGridAreaFromGridAxisSpan`?

Reply:
Fixed